execution.go

```diff
@@ -375,23 +374,32 @@ func extractAndDedupeBoundaryIDs(data interface{}, insertionPoint []string) ([]s
 		deduped = append(deduped, id)
 	}
 
-	return sort.StringSlice(deduped), nil
```

Comment:
Do you think the sort is unnecessary or is it part of the fix? I'd prefer to leave other behaviour changes to a separate PR if possible

Reply:
That's not a function call. It's an unnecessary type cast `type StringSlice []string`. It was never sorting. Do you want it sorted?

Reply:
ah I see, no if it's unnecessary then remove it, it was possibly left over from a refactor.